Fix to use Arduino D0/D1 UART

rolandvs opened the pull request:
Out of the box the UART is not connected to the D0/D1 pins of the Arduino connector. This patch setup UART3 for this task and leave everything as it is.

rolandvs committed:
Fix to use Arduino D0/D1 UART

dpgeorge commented:
Thanks, but I'm not sure about this... I think that the standard configuration for a board should match the hardware that you get out-of-the-box, and not require any hardware mods.  So D0/D1 should stay on PA3/PA2.

rolandvs commented:
They are, but not connected and need a modification always. But is there room to add a readme in the boards/<target> Just to address some specifics/tips for the user?

dpgeorge commented:
> But is there room to add a readme in the boards/ Just to address some specifics/tips for the user?

Yes that is fine to do, and would be useful.

If D0/D1 are really unusable out of the box then maybe we should do something about that...

rolandvs committed:
Restored original board labels

Restored the original definitions

rolandvs committed:
Use README.md to document the use of Arduino D0/D1 UART

Document the alternative use of UART(3) while maintaining UART(2) to the ST-Link VCOM

rolandvs committed:
Fix to use Arduino D0/D1 UART


Restored original board labels

Restored the original definitions 
Use README.md to document the use of Arduino D0/D1 UART

Document the alternative use of UART(3) while maintaining UART(2) to the ST-Link VCOM

rolandvs committed:
Merge remote-tracking branch 'origin/ArduinoUART' into ArduinoUART
